demo/midgard-sdk/src/user-events/withdrawal.ts

```diff
@@ -1,7 +1,48 @@
-import { LucidEvolution, TxBuilder } from "@lucid-evolution/lucid";
+import { AddressData, AddressSchema, HashingError, LucidError, POSIXTime, POSIXTimeSchema, hashHexWithBlake2b256 } from "@/common.js";
+import { WithdrawalBody, WithdrawalEventSchema, WithdrawalInfo, WithdrawalSignature } from "@/ledger-state.js";
+import { CML, Data, LucidEvolution, Script, toUnit, TxBuilder, UTxO } from "@lucid-evolution/lucid";
+import { Effect } from "effect";
 
-export type WithdrawalOrderParams = {};
+export type WithdrawalOrderParams = {
+  withdrawalScriptAddress: string;
+  mintingPolicy: Script;
+  policyId: string;
+  inclusionTime: POSIXTime
+  withdrawalBody : WithdrawalBody;
+  withdrawalSignature: WithdrawalSignature;
+  refundAddress: AddressData;
+  refundDatum: string;
+};
+
+export const WithdrawalOrderDatumSchema = Data.Object({
+  event: WithdrawalEventSchema,
+  inclusionTime: POSIXTimeSchema,
+  refundAddress: AddressSchema,
+  refundDatum: Data.Nullable(Data.Bytes()),
+});
+export type WithdrawalOrderDatum = Data.Static<typeof WithdrawalOrderDatumSchema>;
+export const WithdrawalOrderDatum = WithdrawalOrderDatumSchema as unknown as WithdrawalOrderDatum;
 
+export const WithdrawalMintRedeemerSchema = Data.Enum([
+  Data.Object({
+    AuthenticateEvent: Data.Object({
+      nonceInputIndex: Data.Integer(),
+      eventOutputIndex: Data.Integer(),
+      hubRefInputIndex: Data.Integer(),
+      witnessRegistrationRedeemerIndex: Data.Integer(),
+    }),
+  }),
+  Data.Object({
+    BurnEventNFT: Data.Object({
+      nonceAssetName: Data.Bytes(),
+      witnessUnregistrationRedeemerIndex: Data.Integer(),
+    }),
+  }),
+]);
+export type WithdrawalMintRedeemer = Data.Static<typeof WithdrawalMintRedeemerSchema>;
+export const WithdrawalMintRedeemer =
+  WithdrawalMintRedeemerSchema as unknown as WithdrawalMintRedeemer;
```

Comment:
We are at a point where we can abstract this into a single redeemer (similar to the onchain code), defined in `midgard-sdk/src/user-events/index.ts`.